### ISSUE TITLE
Error in find_rainflow_ranges for boundary cases

### ISSUE THREAD

**Issue by abhijeet-pandey** [Gunnstein/fatpack#14]
find_rainflow_ranges gives error for arbitrary numpy array. For example if the array doesn't contain any cycles  - fatpack.find_rainflow_ranges(np.array([1,2,3,4,5]), return_means=True, k=2**14) produces error.

![image](https://user-images.githubusercontent.com/24558167/197655422-fa620696-7bd9-4aed-ad8e-dd194f73cc38.png)

**Comment by Gunnstein** [comment]
Hey @abhijeet-pandey,

Since you mention "arbitrary": Are there more cases which gives error than if an array does not contain any cycles?

For the case of no cycles in the array, there should at the very least be a different error (FatpackError?) and more descriptive error message. A better solution would perhaps be if an empty array was returned since there are no cycles. I will think a bit more on this issue, but I welcome any help or comments.

Best regards,
Gunnstein

**Comment by abhijeet-pandey** [comment]
I agree, that in case of no cycles - it should return empty array or None.
Yes, I did observe this error on a long data set (gigabyte size) that I had which is hard to share. One characteristic of the data was that it had sections with no cycles. Hence, I tried fatpack with artificial data with no cycles and was able to reproduce the same error. I hacked myself out of this error by using a try-except block over the if statement throwing the error and do nothing (do a "pass") in case of an exception. This may not be a correct approach though.

**Comment by wweijtje** [comment]
Just thinking out loud but would a strain history of `np.array([1,2,3,4,5]` result in a single residual/half cycle as there is a cycle initiated but not closed.

**Comment by abhijeet-pandey** [comment]
Half cycle as per the pagoda method.

**Comment by dsghrg** [comment]
Hello @Gunnstein 

I ran into the same issue with having an `IndexError` mentioned by @abhijeet-pandey but in the function `fatpack.find_reversals`.

As you already mentioned it would be a good idea to have a custom FatPack error, we could have something like this:

```python 
class FatpackError(Exception):
    """Custom exception for fatpack related errors."""
    pass

# snipped that causes errors:
    try:
        # This condition checks if the last point should be included as a reversal
        if (z[revix[-1]] - z[revix[-2]]) * (z[ix[-1]] - z[revix[-1]]) < 0.:
            revix = np.append(revix, ix[-1])
    except IndexError:
        # This error occurs if `revix` has fewer than 2 elements,
        # meaning no turning points were found after the initial point.
        raise FatpackError(
            "Could not determine the final reversal point. "
            "The signal may be too short or have too few turning points "
            "after discretization."
        )
```

Is this what you though? Let me know if you want me to create a pull-request.
